demo environments missing argument "sendgrid_password"

## Problem statement
Qian (Jim) Fu [HDH0]
  [14 hours ago](https://semanticbits.slack.com/archives/C03HG0V4TRC/p1695680755644009)
Experts, I tried to test out some github action changes on demo1, and it failed on "Provision demo envirenment"
If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
Run terraform validate
```
╷
│ Error: Missing required argument
│ 
│   on main.tf line 215, in module "metabase":
│  215: module "metabase" ***
│ 
│ The argument "sendgrid_password" is required, but no definition was found.
╵
```
Error: Process completed with exit code 1.
Run sleep 300;
ERROR: (ResourceNotFound) The Resource 'Microsoft.DBforPostgreSQL/servers/pdhdemo1-pgsql' under resource group 'prime-data-hub-demo1' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix
Code: ResourceNotFound
Message: The Resource 'Microsoft.DBforPostgreSQL/servers/pdhdemo1-pgsql' under resource group 'prime-data-hub-demo1' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix
Error: Process completed with exit code 3.

https://semanticbits.slack.com/archives/C03HG0V4TRC/p1695680755644009

--- a/operations/app/terraform/modules/init/key_vault.tf
+++ b/operations/app/terraform/modules/init/key_vault.tf
@@ -19,6 +19,10 @@ locals {
     cyberark-ip-ingress = {
       secret = "158.111.21.0/24,158.111.123.0/25",
       vault  = azurerm_key_vault.init["keyvault"]
+    },
+    sendgrid-password = {
+      secret = "changeIT!",
+      vault  = azurerm_key_vault.init["appconfig"]
     }
   }
   keys = {

--- a/operations/app/terraform/vars/demo/data.tf
+++ b/operations/app/terraform/vars/demo/data.tf
@@ -10,6 +10,15 @@ data "azurerm_key_vault" "app_config" {
   ]
 }
 
+data "azurerm_key_vault_secret" "sendgrid_password" {
+  name         = "sendgrid-password"
+  key_vault_id = data.azurerm_key_vault.app_config.id
+
+  depends_on = [
+    module.init
+  ]
+}
+
 data "azurerm_key_vault_secret" "postgres_user" {
   name         = "functionapp-postgres-user"
   key_vault_id = data.azurerm_key_vault.app_config.id

--- a/operations/app/terraform/vars/demo/main.tf
+++ b/operations/app/terraform/vars/demo/main.tf
@@ -227,6 +227,7 @@ module "metabase" {
   postgres_server_name   = module.database.postgres_server_name
   postgres_user          = data.azurerm_key_vault_secret.postgres_user.value
   postgres_pass          = data.azurerm_key_vault_secret.postgres_pass.value
+  sendgrid_password      = data.azurerm_key_vault_secret.sendgrid_password.value
   subnets                = module.network.subnets
 }
 

